services/py-api/src/database/repository/participants_repository.py

```diff
@@ -0,0 +1,54 @@
+from typing import Optional, Type
+
+from bson import ObjectId
+from motor.motor_asyncio import AsyncIOMotorClientSession
+from pydantic import BaseModel
+from pymongo.errors import DuplicateKeyError
+from result import Result, Ok, Err
+from structlog.stdlib import get_logger
+
+from src.database.db_manager import DatabaseManager
+from src.database.model.participant_model import Participant
+from src.database.repository.base_repository import CRUDRepository
+from src.server.exception import DuplicateEmail
+from src.server.schemas.request_schemas.schemas import ParticipantRequestBody
+
+LOG = get_logger()
+
+
+class ParticipantsRepository(CRUDRepository):
+    # TODO: Implement the rest of the methods
+    def __init__(self, db_manager: DatabaseManager, collection: str) -> None:
+        self._client = db_manager.client
```

Comment:
Probably a good idea is to have a `get_client` and `get_collection` methods and define the client as a private member of the `db_manager`

Reply:
It is defined as a private member?

Reply:
I'm talking about the `db_manager.client` and the collection fetch you do the line below

Reply:
The db client is a private member of the DatabaseManager and`db_manager.client` is defined as a property (getter):
<img width="432" alt="image" src="https://github.com/user-attachments/assets/4bb3f576-d3ec-4301-b261-6f368b00d200">

<img width="524" alt="image" src="https://github.com/user-attachments/assets/460f874c-e004-4206-9060-4de02f7ec081">

Should I rename it to get_client?

The `self._collection = self._client[db_manager.DB_NAME][collection]` on the next line is required as every repository is responsible for working with a particular collection, and in order for us to create an ` AsyncIOMotorCollection` we need the client. I can do it like this for simplicity:
```python
self._collection = db_manager.client[db_manager.DB_NAME][collection]
```
I am not sure we need a getter for the collection itself.

Reply:
Fixed in 35cfe383f588e86496b7f3e3fb8e855965ebe35a

Now it is a better interface indeed, thank you!